### ISSUE TITLE
Teacher-forcing in the Transformer tutorial

### ISSUE THREAD

**Issue by UltraSpecialException** [tensorflow/tensorflow#30852]
Thank you for submitting a TensorFlow documentation issue. Per our GitHub
policy, we only address code/doc bugs, performance issues, feature requests, and
build/installation issues on GitHub.

The TensorFlow docs are open source! To get involved, read the documentation
contributor guide: https://www.tensorflow.org/community/contribute/docs

Please provide a link to the documentation entry, for example:
https://www.tensorflow.org/beta/tutorials/text/transformer#training_and_checkpointing

## Description of issue (what needs changing):
Teacher-forcing seems to not be implemented?

### Clear description
The documentation here mentions that the training uses teacher-forcing, however, it doesn't seem like, with the code shown, that this is implemented. The variable `tar_real` is the true outputs, but it seems to only be used for loss and accuracy computations?

Please let me know if I'm making a mistake here! Thanks in advance.

**Comment by tlkh** [comment]
Each `train_step` takes in `inp` and `tar` objects from the dataset in the training loop. Teacher forcing is indeed used since the correct example from the dataset is always used as input during training (as opposed to the "incorrect" output from the previous training step): 

* `tar` is split into `tar_inp`, `tar_real` (offset by one character)
* `inp`, `tar_inp` is used as input to the model
* model produces an output which is compared with `tar_real` to calculate loss
* model output is discarded (not used anymore)
* repeat loop

> Teacher forcing is a procedure ... in which during training the model receives the ground truth output y(t) as input at time t+1.
> Page 372, Deep Learning, 2016.

**Comment by UltraSpecialException** [comment]
@tlkh Thanks for the response! I see what you mean.
